cli/command/context/cmd.go

```diff
@@ -28,6 +29,13 @@ func NewContextCommand(dockerCli command.Cli) *cobra.Command {
 	return cmd
 }
 
-func normalizeContextName(name string) string {
-	return strings.ToLower(name)
+const restrictedNamePattern = "^[a-zA-Z0-9][a-zA-Z0-9_.+-]+$"
```

Comment:
Where did the `+` come from?

Note that this pattern still allows names such as `foo......bar` or `foo.....`; this is currently the case for containers as well, and may not be a concern; just calling it out here in case we want to be more conservative (e.g., only accept `_`, `.`, -` as a separator between alphanumeric characters, and only a single consecutive separator

Reply:
I added the +, because as context are the aggregation of swarm and kubernetes endpoint configurations, I'd personnally like to name my context like `local-swarm+aks-west-eu`. Also, with the dir names being now the sha256 of the context names, I have no problem if a user might think that `fo.....` is a valuable context name